### ISSUE TITLE
vcpkg fails to install openssl due to hash check failure

### ISSUE THREAD

**Issue by Jaysinh** [Microsoft/vcpkg#1189]
tried to install RESTC++ library from https://github.com/Microsoft/cpprestsdk. However fails to install due to the following error where the openssl download failes the hash check. 
```
`C:\Users\Jaysinh\Documents\workspace\vcpkg>vcpkg install openssl
The following packages will be built and installed:
    openssl:x86-windows
Building package openssl:x86-windows...
-- CURRENT_INSTALLED_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/installed/x86-windows
-- DOWNLOADS=C:/Users/Jaysinh/Documents/workspace/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/packages/openssl_x86-windows
-- CURRENT_BUILDTREES_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/buildtrees/openssl
-- CURRENT_PORT_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/ports/openssl/.
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:149 (file):
  file DOWNLOAD HASH mismatch

    for file: [C:/Users/Jaysinh/Documents/workspace/vcpkg/downloads/strawberry-perl-5.24.1.1-32bit-portable.zip]
      expected hash: [a6e685ea24376f50db5f06c5b46075f1d3be25168fa1f27fa9b02e2ac017826cee62a2b43562f9b6c989337a231ba914416c110075457764de2d11f99d5e0f26]
        actual hash: [f3355d2b16232e34d28713216e7f49fae02c6b8e957548fd0eda19a3894e43c4ad613acc0685c86f450de36338cf9681d8619c47bd11c147a9079f300d83e40f]
             status: [0;"No error"]

Call Stack (most recent call first):
  ports/openssl/portfile.cmake:9 (vcpkg_find_acquire_program)
  scripts/ports.cmake:73 (include)


Error: Building package openssl:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openssl:x86-windows
  Vcpkg version: 0.0.81-39b69e5b06df309b9dcfea55296a4c2c3eea93ce

Additionally, attach any relevant sections from the log files above.`
```

**Comment by ras0219-msft** [comment]
Hmm, this looks like https://github.com/Microsoft/vcpkg/issues/932, though your reported status is [0;"No error"] instead of [56;"Failure when receiving data from the peer"].

Could you try deleting the downloaded file and re-running vcpkg install? Does it fail with the same status (No error) and hash?

**Comment by Jaysinh** [comment]
Hi, I deleted the downloaded files and re ran the command, but I get the same error. 

```
The following packages will be built and installed:
  * boost:x86-windows
  * boost:x64-windows
  * bzip2:x86-windows
  * bzip2:x64-windows
    cpprestsdk:x86-windows
    cpprestsdk:x64-windows
  * openssl:x86-windows
  * openssl:x64-windows
  * websocketpp:x86-windows
  * websocketpp:x64-windows
Additional packages (*) will be installed to complete this operation.
Building package openssl:x86-windows...
-- CURRENT_INSTALLED_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/installed/x86-windows
-- DOWNLOADS=C:/Users/Jaysinh/Documents/workspace/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/packages/openssl_x86-windows
-- CURRENT_BUILDTREES_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/buildtrees/openssl
-- CURRENT_PORT_DIR=C:/Users/Jaysinh/Documents/workspace/vcpkg/ports/openssl/.
-- [download 0% complete]
-- [download 1% complete]
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:149 (file):
  file DOWNLOAD HASH mismatch

    for file: [C:/Users/Jaysinh/Documents/workspace/vcpkg/downloads/strawberry-perl-5.24.1.1-32bit-portable.zip]
      expected hash: [a6e685ea24376f50db5f06c5b46075f1d3be25168fa1f27fa9b02e2ac017826cee62a2b43562f9b6c989337a231ba91441
6c110075457764de2d11f99d5e0f26]
        actual hash: [ee82a9677fa94b6ea71b7ee6bfcac6d512eef1b6e9083e07f8bee4e791cc8f36386d5701d961ff6806c50895eecb953aa3
7ee045c808500703b2bcae6db394bd]
             status: [56;"Failure when receiving data from the peer"]

Call Stack (most recent call first):
  ports/openssl/portfile.cmake:9 (vcpkg_find_acquire_program)
  scripts/ports.cmake:73 (include)


Error: Building package openssl:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openssl:x86-windows
  Vcpkg version: 0.0.81-39b69e5b06df309b9dcfea55296a4c2c3eea93ce

Additionally, attach any relevant sections from the log files above.
```

**Comment by ras0219-msft** [comment]
Thanks! It looks like something is causing the connection to reset.

Could you try downloading the zip file and manually placing it in the downloads folder as a workaround?

http://strawberryperl.com/download/5.24.1.1/strawberry-perl-5.24.1.1-32bit-portable.zip

We'll still hash check it, so it won't negatively impact anything else.

**Comment by Jaysinh** [comment]
on trying to download it, its blocked as there is a trojan detected in it :/

**Comment by alexbryant** [comment]
Hi, I am having the same issue only it seems the old strawberry perl site is down. Is there another safe source for downloading it?

**Comment by atkawa7** [comment]
@alexbryant Try https://web.archive.org/web/20170614020009/http://strawberryperl.com/download/5.24.1.1/strawberry-perl-5.24.1.1-32bit-portable.zip

**Comment by ruizpauker** [comment]
I'm also having this issue, the file "strawberry-perl-5.24.1.1-32bit-portable.zip" is being blocked by our Firewalls, claiming it contains a Trojan.

I've tested downloading the latest 5.24.x version: strawberry-perl-5.24.3.1-32bit-portable.zip, it was successfully downloaded.

That said, IMHO the solution to this issue is to upgrade the StrawberryPerl version to 5.24.3.1, this should be done in: ./scripts/cmake/vcpkg_find_acquire_program.cmake:52.

I'll try it and let you know.

**Comment by NancyLi1013** [comment]
Please open a new issue if this is still a problem for you.
